src/libraries/Common/src/Interop/OSX/Interop.Network.Tls.cs

```diff
@@ -0,0 +1,153 @@
+
+            [LibraryImport(Interop.Libraries.AppleNetworkNative, EntryPoint = "AppleNetNative_NwSetTlsOptions", StringMarshalling = StringMarshalling.Utf8)]
+            private static partial int SetTlsOptions(SafeNwHandle connection, IntPtr gcHandle,
+                                                            string targetName, Span<byte> alpnBuffer, int alpnLength,
+                                                            SslProtocols minTlsProtocol, SslProtocols maxTlsProtocol);
+
+            internal static int SetTlsOptions(SafeNwHandle nwHandle, IntPtr gcHandle, string targetName, List<SslApplicationProtocol>? applicationProtocols, SslProtocols minTlsVersion, SslProtocols maxTlsVersion)
+            {
+                int alpnLength = GetAlpnProtocolListSerializedLength(applicationProtocols);
+                Span<byte> alpn = alpnLength <= 256 ? stackalloc byte[256].Slice(0, alpnLength) : new byte[alpnLength];
+                SerializeAlpnProtocolList(applicationProtocols, alpn);
+
+                return SetTlsOptions(nwHandle, gcHandle, targetName, alpn, alpnLength, minTlsVersion, maxTlsVersion);
+            }
+
+            [LibraryImport(Interop.Libraries.AppleNetworkNative, EntryPoint = "AppleNetNative_NwStartTlsHandshake")]
+            internal static partial int StartTlsHandshake(SafeNwHandle connection, IntPtr gcHandle);
+
+            [LibraryImport(Interop.Libraries.AppleNetworkNative, EntryPoint = "AppleNetNative_NwProcessInputData")]
+            internal static unsafe partial int ProcessInputData(SafeNwHandle connection,
+                                                               SafeNwHandle framer,
+                                                               byte* buffer, int bufferLength);
+
+            [LibraryImport(Interop.Libraries.AppleNetworkNative, EntryPoint = "AppleNetNative_NwSendToConnection")]
+            internal static unsafe partial int SendToConnection(SafeNwHandle connection, IntPtr gcHandle,
+                                                               void* buffer, int bufferLength);
+
+            [LibraryImport(Interop.Libraries.AppleNetworkNative, EntryPoint = "AppleNetNative_NwReadFromConnection")]
+            internal static partial int ReadFromConnection(SafeNwHandle connection, IntPtr gcHandle);
+
+            [LibraryImport(Interop.Libraries.AppleNetworkNative, EntryPoint = "AppleNetNative_NwCancelConnection")]
+            internal static partial int CancelConnection(SafeNwHandle connection);
+
+            [LibraryImport(Interop.Libraries.AppleNetworkNative, EntryPoint = "AppleNetNative_NwGetConnectionInfo")]
+            internal static unsafe partial int GetConnectionInfo(SafeNwHandle connection,
+                                                               out SslProtocols pProtocol, out TlsCipherSuite pCipherSuiteOut,
+                                                               ref void* negotiatedAlpn, out uint alpnLength);
```

Comment:
Is alpnLength the length of negotiatedAlpn? Typically it's helpfulf for a buffer ptr and associated length to be named in a corresponding way, e.g. foo and fooLength, so negotiatedAlpn and negotatiedAlpnLength.

Also, why is this void* but other APIs are using byte*. Is this one not bytes?

And why is this length uint when other lengths are int?